### PR TITLE
KAFKA-8744: Update Scala API to give names to processors

### PR DESCRIPTION
**Pull request by mdespriee**
* As it's only API extension to match the java API with Named object with lots of duplication, I only tested the logic once.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mdespriee** [comment]
Failing tests looks like flaky tests. Seems unrelated to this change.

**Comment by bbejeck** [comment]
@mdespriee Thanks for your patience; I'm taking a look today

**Review by bbejeck — COMMENTED** [review]
Thanks for taking this on @mdespriee.  Overall this looks good to me.  I understand the rationale for only one test, but can we add 2-3 more? I'm thinking one `KTable` and maybe one for a join and aggregation.

**Comment by bbejeck** [comment]
ping @mdespriee feature freeze for 2.8 is approaching it would be great to get this in.

**Comment by mdespriee** [comment]
@bbejeck of course. Just rebased on trunk and added a couple of tests.

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/kstream/KTableTest.scala [hunk 1]
```diff
@@ -391,4 +390,57 @@ class KTableTest extends FlatSpec with Matchers with TestDriver {
...
+    val sourceTopic = "source"
+    val sinkTopic = "sink"
+
+    val table = builder.stream[String, String](sourceTopic).groupBy((key, _) => key).count(Named.as("my-name"))
+    table.toStream.to(sinkTopic)
+
+    import scala.jdk.CollectionConverters._
+
+    val filterNode = builder.build().describe().subtopologies().asScala.toList(1).nodes().asScala.toList(1)
+    filterNode.name() shouldBe "my-name"
```
Comment (bbejeck) [inline streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/kstream/KTableTest.scala]: shouldn't this line verify the `countNode` name?

**Review by bbejeck — COMMENTED** [review]
Thanks for the update @mdespriee I just have one minor question, otherwise, this looks good to me.

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/kstream/KTableTest.scala (hunk 1 above)
Comment (mdespriee) [reply streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/kstream/KTableTest.scala]: bad copy-paste of variable name.

**Review by bbejeck — APPROVED** [review]
LGTM! 

I'll merge once the build completes.

**Comment by bbejeck** [comment]
Java 11 and 15 passed

Java 8 failures unrelated
```
org.apache.kafka.clients.consumer.KafkaConsumerTest.testCloseWithTimeUnit()
org.apache.kafka.clients.consumer.internals.FetcherTest.testEarlierOffsetResetArrivesLate()
kafka.api.TransactionsBounceTest.testWithGroupMetadata()
```
Kicking off tests again to try and get a green build

**Comment by bbejeck** [comment]
merged #9738 into trunk

**Comment by bbejeck** [comment]
Thanks again @mdespriee for the contribution!

**Comment by mdespriee** [comment]
@bbejeck you're welcome, thanks for the review and merge :-)

**Pull request merged**